src/nomad_simulations/properties/band_structure.py

```diff
@@ -0,0 +1,370 @@
+
+    def normalize(self, archive, logger) -> None:
+        super().normalize(archive, logger)
+
+
+class ElectronicEigenvalues(BaseElectronicEigenvalues):
+    """ """
+
+    iri = 'http://fairmat-nfdi.eu/taxonomy/ElectronicEigenvalues'
+
+    occupation = Quantity(
+        type=np.float64,
+        shape=['*', 'n_bands'],
+        description="""
+        Occupation of the electronic eigenvalues. This is a number between 0 and 2, where 0 means
+        that the state is unoccupied and 2 means that the state is fully occupied. It is controlled
+        by the Fermi-Dirac distribution:
+
+            $ f(E) = 1 / (1 + exp((E - E_F) / kT)) $
+
+        The shape of this quantity is defined as `[KMesh.n_points, KMesh.dimensionality, n_bands]`, where `KMesh` is a `variable`.
+        """,
+    )
+
+    highest_occupied = Quantity(
+        type=np.float64,
+        unit='joule',
+        description="""
+        Highest occupied electronic eigenvalue. Together with `lowest_unoccupied`, it defines the
+        electronic band gap.
+        """,
+    )
+
+    lowest_unoccupied = Quantity(
+        type=np.float64,
+        unit='joule',
+        description="""
+        Lowest unoccupied electronic eigenvalue. Together with `highest_occupied`, it defines the
+        electronic band gap.
+        """,
+    )
+
+    value_contributions = SubSection(
+        sub_section=BaseElectronicEigenvalues.m_def,
+        repeats=True,
+        description="""
+        Contributions to the electronic eigenvalues. Example, in the case of a DFT+GW calculation, the GW eigenvalues
+        are stored under `value`, and each contribution is identified by `label`:
+            - `'KS'`: Kohn-Sham contribution. This is also stored in the DFT entry under `ElectronicEigenvalues.value`.
+            - `'KSxc'`: Diagonal matrix elements of the expectation value of the Kohn-Sahm exchange-correlation potential.
+            - `'SigX'`: Diagonal matrix elements of the exchange self-energy. This is also stored in the GW entry under `ElectronicSelfEnergy.value`.
+            - `'SigC'`: Diagonal matrix elements of the correlation self-energy. This is also stored in the GW entry under `ElectronicSelfEnergy.value`.
+            - `'Zk'`: Quasiparticle renormalization factors contribution. This is also stored in the GW entry under `QuasiparticleWeights.value`.
+        """,
+    )
+
+    reciprocal_cell = Quantity(
+        type=KSpace.reciprocal_lattice_vectors,
+        description="""
+        Reference to the reciprocal lattice vectors stored under `KSpace`. This reference is useful when resolving the Brillouin zone
+        for the front-end visualization.
+        """,
+    )
+
+    def __init__(
+        self, m_def: Section = None, m_context: Context = None, **kwargs
+    ) -> None:
+        super().__init__(m_def, m_context, **kwargs)
+        self.name = self.m_def.name
+
+    def _check_occupation_shape(self, logger: BoundLogger) -> bool:
```

Comment:
not sure why you are passing logger? i would define it globally LOGGER = utils.get_logger(__name__)

Reply:
Yes, this is confusing for me. As I am passing logger to `normalize()` would these conflict? I thought that I had to keep the same `logger` in all places to write the messages in the same bound logger.

Reply:
ah yeah right then ignore my comment.